lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/EvaluatorRuntime.cs

```diff
@@ -254,21 +253,12 @@ private void OnException(Exception e)
 
                 try
                 {
-                    using (var memStream = new MemoryStream())
-                    {
-                        var formatter = new BinaryFormatter();
-                        formatter.Serialize(memStream, e);
-                        errorBytes = memStream.ToArray();
-                    }
+                    errorBytes = ByteUtilities.SerializeToBinaryFormat(e);
                 }
                 catch (SerializationException se)
```

Comment:
Sorry, I was not clear. We could move the whole business of trying to serialize and to use a `NonSerializableEvaluatorException` when that fails in a utility method. This same `try/catch` is repeated several times in the code.